src/NServiceBus.TransportTests/When_failure_happens_after_isolated_send.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace NServiceBus.TransportTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+    using Transport;
+
+    public class When_failure_happens_after_isolated_send : NServiceBusTransportTest
+    {
+        [TestCase(TransportTransactionMode.None)]
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task Should_emit_messages(TransportTransactionMode transactionMode)
+        {
+            var onMessageCalled = new TaskCompletionSource<bool>();
+
+            OnTestTimeout(() => onMessageCalled.SetCanceled());
+
+            await StartPump(async context =>
+                {
+                    if (context.Headers.ContainsKey("CompleteTest"))
+                    {
+                        onMessageCalled.SetResult(false);
+                        return;
+                    }
+
+                    if (context.Headers.ContainsKey("IsolatedSend"))
+                    {
+                        onMessageCalled.SetResult(true);
+                        return;
+                    }
+
+                    await SendMessage(InputQueueName, new Dictionary<string, string>
+                    {
+                        {"IsolatedSend", "true"}
+                    }, context.TransportTransaction, null, DispatchConsistency.Isolated);
+
+                    throw new Exception("Simulated exception");
+                },
+                async context =>
+                {
+                    await SendMessage(InputQueueName, new Dictionary<string, string>
```

Comment:
looked at the code with @dbelcham and it does indeed look like this message even causes problems as we will set the result on the TCS multiple times. Since the done condition is already met, we don't see that we will resend this message endlessly (can be reproduced by adding a Task.Delay after the `await SendMessage(InputQueueName);` statement.

@dbelcham is going to address this